continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerArchiver.java

```diff
@@ -23,47 +23,41 @@
 import java.util.function.BiPredicate;
 import java.util.regex.Pattern;
 import java.util.stream.Stream;
-import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
-@RequiredArgsConstructor
 class ContinuousAsyncProfilerArchiver implements Runnable {
     private final ContinuousAsyncProfilerProperties properties;
+    private final Path continuousDir;
+    private final BiPredicate<Path, BasicFileAttributes> predicate;
 
-    @Override
-    @SuppressWarnings("BusyWait")
-    public void run() {
-        Path continuousDir = Paths.get(properties.getContinuousOutputDir());
+    public ContinuousAsyncProfilerArchiver(ContinuousAsyncProfilerProperties properties) {
+        this.properties = properties;
+        this.continuousDir = Paths.get(properties.getContinuousOutputDir());
         Pattern compiledPattern = Pattern.compile(properties.getArchiveCopyRegex());
```

Comment:
it would be better to compile it once (then cache) inside properties then just pass pattern in constructor

Reply:
done